File: typescript-connector/src/models/Volume.ts
// Class from DFC Business Ontology: #Volume
import { SemanticObject } from "../core/SemanticObject.js";
import { QuantitativeValue, type QuantitativeValueParams } from "./QuantitativeValue.js";

export interface VolumeParams extends QuantitativeValueParams {}
export class Volume extends QuantitativeValue {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Volume";
  }



  constructor(
    semanticId: string,
    params?: VolumeParams,
  ) {
    super(semanticId, params);
    this.semanticType = Volume.SEMANTIC_TYPE;
  }
  static {
    SemanticObject.typeRegistry.set(Volume.SEMANTIC_TYPE, Volume);
  }
}

File: ruby-gem/lib/models/volume.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Volume
require_relative '../core/semantic_object'
require_relative 'quantitative_value'


module DfcLinkmlConnector
  module Models
    class Volume < QuantitativeValue
      SEMANTIC_TYPE = "dfc-b:Volume".freeze

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, value: nil, unit: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, value: nil, unit: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension, value: value, unit: unit)

        self.semanticType = "dfc-b:Volume"

      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
